src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestExpressions.ll

```diff
@@ -0,0 +1,72 @@
+entry:
+  %0 = call %Callable* @Microsoft__Quantum__Testing__QIR__ReturnGlobalId__body()
+  %1 = call %Callable* @Microsoft__Quantum__Testing__QIR__ReturnLocalId__body()
+  %2 = call i64 @Microsoft__Quantum__Testing__QIR__ReturnFunctionCall__body()
+  %3 = call double @Microsoft__Quantum__Testing__QIR__ReturnOperationCall__body()
+  %4 = call %Callable* @Microsoft__Quantum__Testing__QIR__ReturnPartialApplication__body()
+  %5 = call { %TupleHeader, { %TupleHeader, i2, i64 }*, double }* @Microsoft__Quantum__Testing__QIR__ReturnUnwrapApplication__body()
+  %6 = call %Callable* @Microsoft__Quantum__Testing__QIR__ReturnAdjointApplication__body()
+  %7 = call %Callable* @Microsoft__Quantum__Testing__QIR__ReturnControlledApplication__body()
+  %8 = call { %TupleHeader, i64, { %TupleHeader, %Callable*, %String* }* }* @Microsoft__Quantum__Testing__QIR__ReturnTuple__body()
+  %9 = call %String* @Microsoft__Quantum__Testing__QIR__ReturnArrayItem__body()
+  %10 = call i2 @Microsoft__Quantum__Testing__QIR__ReturnNamedItem__body()
+  %11 = call %Array* @Microsoft__Quantum__Testing__QIR__ReturnArray__body()
+  %12 = call %Array* @Microsoft__Quantum__Testing__QIR__ReturnNewArray__body()
+  %13 = call %String* @Microsoft__Quantum__Testing__QIR__ReturnString__body()
+  %14 = call %Range @Microsoft__Quantum__Testing__QIR__ReturnRange__body()
+  %15 = call %Array* @Microsoft__Quantum__Testing__QIR__ReturnCopyAndUpdateArray__body()
+  %16 = call { %TupleHeader, { %TupleHeader, i2, i64 }*, double }* @Microsoft__Quantum__Testing__QIR__ReturnCopyAndUpdateUdt__body()
```

Comment:
why these TupleHeader's don't need to be unreferenced (unlike %8)?

Reply:
I am working on some changes that will add an unreference to %16. However, I would prefer to make this a separate PR.